hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java

```diff
@@ -63,102 +54,22 @@
  */
 @UseRecordReaderFromInputFormat
 @UseFileSplitsFromInputFormat
-public class HoodieParquetInputFormat extends MapredParquetInputFormat implements Configurable {
+public class HoodieParquetInputFormat extends HoodieFileInputFormatBase implements Configurable {
```

Comment:
this is a problem. Hive code special cases subclasses of  `MapredParquetInputFormat` for some of the optimizations. So we need to maintain this hierarchy.

Reply:
stuff like this https://github.com/apache/hive/blob/6d8c9aab795e51278bca27f56053e9eccec01e90/ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsNoJobTask.java#L99